api_core/google/api_core/protobuf_helpers.py

```diff
@@ -247,3 +249,41 @@ def setdefault(msg_or_dict, key, value):
     """
     if not get(msg_or_dict, key, default=None):
         set(msg_or_dict, key, value)
+
+
+def fieldmask(original, modified):
+    """Create a field mask by comparing two messages.
+
+    Args:
+        original (~google.protobuf.message.Message): the original message. 
+        modified (~google.protobuf.message.Message): the modified message.
+
+    Returns:
+        google.protobuf.field_mask_pb2.FieldMask: returns a FieldMask instance
+            representing the differences between between two messages.
```

Comment:
So this is a little unclear. The fieldmask doesn't represent the differences, but rather notes *which* fields are different. Perhaps this is a bit more clear:

```
A field mask that contains the list of field names that have different values between the two messages. If the messages are equivalent, then the field mask is empty.
```.